lib/libc/minimal/Kconfig

```diff
@@ -14,29 +14,30 @@ config MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS
 	  option may require an additional memory protection region.
 
 config MINIMAL_LIBC_MALLOC
-	bool "Minimal libc malloc implementation"
+	bool "Legacy alias for COMMON_LIBC_MALLOC"
 	default y
+	select COMMON_LIBC_MALLOC
 	help
 	  Enable the minimal libc's implementation of malloc, free, and realloc.
 	  Disable if you wish to provide your own implementations of these functions.
 
 config MINIMAL_LIBC_MALLOC_ARENA_SIZE
-	int "Size of the minimal libc malloc arena"
+	int "Legacy alias for COMMON_LIBC_MALLOC_ARENA_SIZE"
```

Comment:
This should be deprecated as per https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated

Reply:
That doesn't seem to have explicit instructions on how to deprecate a Kconfig variable, but it looks like there are a couple of patterns I can follow in the existing code.

Reply:
ok, I've added [DEPRECATED] to each affected Kconfig variable, along with references to the new variable.